### PR TITLE
RavenDB-15538: Missing attachment break replication 4.2

### PR DESCRIPTION
**Pull request by garayx**

### CODE REVIEW

**Comment by garayx** [comment]
5.0: https://github.com/ravendb/ravendb/pull/11171

File: src/Raven.Server/Documents/Replication/ReplicationLoader.cs [hunk 1]
```diff
@@ -1550,6 +1551,66 @@ public int GetNextReplicationStatsId()
...
+        internal TestingStuff _forTestingPurposes;
+        internal TestingStuff ForTestingPurposesOnly()
+        {
+            if (_forTestingPurposes != null)
+                return _forTestingPurposes;
+
+            return _forTestingPurposes = new TestingStuff();
+        }
+
+        internal class TestingStuff
```
Comment (karmeli87) [inline src/Raven.Server/Documents/Replication/ReplicationLoader.cs]: Why do we need all this? can't we use `GetReplicationItems` in both dbs?

Comment (garayx) [reply src/Raven.Server/Documents/Replication/ReplicationLoader.cs]: done

File: test/SlowTests/Issues/RavenDB_15538.cs [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+        {
+            if (item.Type == ReplicationBatchItem.ReplicationItemType.Attachment)
+            {
+                item.Stream.Position = 0;
+                var stream = new MemoryStream();
+                item.Stream.CopyTo(stream);
+                stream.Position = 0;
+                item.Stream.Position = 0;
+
+                return new ReplicationBatchItem
```
Comment (karmeli87) [inline test/SlowTests/Issues/RavenDB_15538.cs:268]: For the 5.x please use the `Clone` method

**Review by karmeli87 — APPROVED** [review]

**Pull request merged**